### PR TITLE
Initial version of repo tag dumper

### PR DESCRIPTION
**Pull request by justinsb**
Dumps tags to json for later grepping / import.

Issue #2037

### CODE REVIEW

File: metrics/access-logs/sniped/main.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
+package main
```
Comment (spiffxp) [inline metrics/access-logs/sniped/main.go:1]: Needs boilerplate

File: metrics/access-logs/sniped/go.mod [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module sniped
+
+go 1.14
```
Comment (spiffxp) [inline metrics/access-logs/sniped/go.mod:3]:
```suggestion
go 1.16
```

Unless there's a reason to use 1.14?

File: metrics/access-logs/sniped/main.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+	for _, manifest := range tags.Manifests {
+
+		for _, tag := range manifest.Tags {
+			/*
+				ref, err := name.ParseReference(digest, repo.Registry)
+				if err != nil {
+					return fmt.Errorf("parsing reference %q: %v", r, err)
+				}
+				manifest, err := remote.Get(ref, remoteOptions...)
+			*/
```
Comment (spiffxp) [inline metrics/access-logs/sniped/main.go:109]: Not a fan of checking in commented-out code.  Why was this commented out?  Dropping a TODO would be more helpful
```suggestion
```

File: metrics/access-logs/sniped/main.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+	/*
+		for digest, manifest := range tags.Manifests {
+			fmt.Printf("%s@%s\n", repo, digest)
+
+			for _, tag := range manifest.Tags {
+				fmt.Printf("%s:%s\n", repo, tag)
+			}
+		}
+	*/
```
Comment (spiffxp) [inline metrics/access-logs/sniped/main.go:156]: Same thing re: commented-out code
```suggestion
```

File: metrics/access-logs/sniped/main.go [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+	"k8s.io/klog"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/google"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+// Based on gcrane list command  https://github.com/google/go-containerregistry/blob/master/cmd/gcrane/cmd/list.go
```
Comment (spiffxp) [inline metrics/access-logs/sniped/main.go:19]: Need more a description on what this is doing and why

File: metrics/access-logs/sniped/go.mod [hunk 6]
```diff
@@ -0,0 +1,8 @@
+module sniped
```
Comment (spiffxp) [inline metrics/access-logs/sniped/go.mod:1]: I don't think this should be in `metrics/access-logs`

It doesn't do anything with metrics (read or publish), and it definitely doesn't do anything with access logs

I guess we lack a "useful miscellaneous tooling" directory.  I would either use `audit/` as the home for tools that help audit our project, or make a new `tools/` directory.  Then put `sniped` under that

**Review by spiffxp — CHANGES_REQUESTED** [review]
It looks like there's still a bunch of iteration happening in https://github.com/kubernetes/k8s.io/issues/2037 so I'm not even sure if this intended to land yet, or at all.

I took a pass with a "merge-and-iterate" perspective, these are the things that block.

**Comment by spiffxp** [comment]
Is this used or intended for merge? I'm going to close next week if not.

ref: https://github.com/kubernetes/k8s.io/issues/1834

**Comment by BobyMCbobs** [comment]
> /cc @BobyMCbobs @Riaankl @hh

This might be interesting to refer to again later on. I believe that we're parsing the paths visited from the pii logs to get the image repo and tag

**Pull request closed without merging**